### ISSUE TITLE
PeerDependencies in packageJSON causing npm install errors

### ISSUE THREAD

**Issue by tirthaguha** [marionettejs/backbone.marionette#3608]
### PeerDependencies in packageJSON causing npm install errors

 1. Step to reproduce
     ```npm install underscore//installs v1.9.0``` 
     ```npm install backbone //installs v1.3.3```
     ```npm install backbone.marionette //installs 3.5.1```
     
>Throws annoying error ```-- UNMET PEER DEPENDENCY underscore@1.9.0```
     
 
 2. Proposed Solution
    in package json, change the peer dependency section
    **FROM**
    ```
    "peerDependencies": {
      "backbone": "~1.3.3",
      "underscore": "~1.8.3"
    },
    ```
    **TO**
    ```
    "peerDependencies": {
      "backbone": "^1.3.3",
      "underscore": "^1.8.3"
    },
    ```
### Expected behavior

There should not be any npm install error

### Actual behavior

```-- UNMET PEER DEPENDENCY underscore@1.9.0```

### Environment

1. Marionette version: 3.5.1
2. Backbone version: 1.3.3
3. Underscore version: 1.9.0
4. NodeJS 6.x, NPM 3.x

**Comment by gam6itko** [comment]
It also can be
"peerDependencies": {
  "backbone": "^1.3",
  "underscore": "^1.8"
},

**Comment by tirthaguha** [comment]
suites me. :+1:

**Comment by paulfalgout** [comment]
Previously `Mn` had locked the versions because Backbone does not follow semver, but Marionette does.. so allowing for a breaking `1.9.x` could break an install that uses `^1.8.3`  This is still possible, but I feel less concerned about it if they're defined as peerDependencies.  That said.. unmet peer dependency is a warning is it not?

**Comment by tirthaguha** [comment]
@paulfalgout that was the first thought. however, by not allowing ^ we are kind of forcing the newer applications to run on old underscoreJS version. 
Agree that peerDependency is an warning, but it is annoying enough, when you have a huge ecosystem running on marionette, and the build machines puke an warning with every build or every component.

**Comment by paulfalgout** [comment]
So if a newer version of underscore is breaking and requires a fix in Marionette, what would we do?

Let's layout a scenario
Mn at v5.5.1 supports underscore ^2.1.1
Underscore 2.2.0 is released and unfortunately includes a breaking change such that v5.5.1 doesn't work without updates.  Underscore does not consider this to be a breaking semver change so it is up to Marionette to fix it.
We have few options.  Since 2.2.0 is breaking, to support it we need to roll a major release of Marionette, but since v.5.5.1 already supports `^2` to fix `v5` we need to roll `v.5.5.2` which changes it to `~2.1.1` so that `2.2.0` is never loaded.

We used to keep a support range of the library and would not update it unless it was proven not to include breaking changes.  But this prevented anyone from using a newer version until Mn updated.

Or we can put the peerDependency to the lowest tested library which doesn't prevent installation of newer versions and updates are up to the dev with the warning?

As it is currently 1.9.0 is bug ridden and 1.9.1 seems to have stalled so we haven't been particularly incentivized to officially support it with Mn.  If it gets better I don't think the correct call is `^`.  The warning is good until the supported non-breaking range can be officially modified with a release, but  keeping it in the peerDependencies allows users to get ahead of the official support if they're willing to deal with the warning.

**TL;DR** The warning is better than allowing breaking changes on dependency updates that do not follow semver.

**Comment by tirthaguha** [comment]
@paulfalgout you put a range in your peer dependencies. underscore is never a Marionette dependency as far as I know, It is a BackboneJS dependency. So, if backbone is comfortable with higher versions of underscore, Marionette shouldn't have a problem.
Marionette, can keep backbone.radio in its peer dependency, but keeping underscore is wrong.

**Comment by paulfalgout** [comment]
Marionette uses underscore everywhere https://github.com/marionettejs/backbone.marionette/blob/master/src/object.js#L4

Also backbone can be comfortable with higher versions of underscore because it doesn't follow semver.  Backbone 1.4.0 can be entirely breaking from 1.3.1 and that is ok with them.  Marionette is not.  We follow semver.. and so does backbone.radio

**Comment by JSteunou** [comment]
I'm closing this, see #3609 for the reasons
